core/classes/Adapter.js

```diff
@@ -200,7 +217,79 @@ module.exports = class Adapter {
       }
     })
 
-    return trx ? query.transacting(trx) : query
+    let result = await (trx ? query.transacting(trx) : query)
+    for (let field of otmPopulate) {
+      await this._oneToModelPopulate(result, field)
+    }
+    for (let field of mtmPopulate) {
+      await this._modelToModelPopulate(result, field.linkerModel, field.initialModel, field.modelToLink)
+    }
+    return result
+  }
+
+  /**
+   * Populates the given models with 1:m relation
+   * @param {Object[]} initialModelArr The array of models to be populated
+   * @param {String} populatingModelIdentity The identity of related model
+   * @private
+   */
+  async _oneToModelPopulate(initialModelArr, populatingModelIdentity) {
+    let idsSet = new Set()
+    for (let model of initialModelArr) {
+      idsSet.add(model[populatingModelIdentity])
+    }
+    const idsArr = [...idsSet]
+
+    const populatingModelsArr = await this.knex(populatingModelIdentity.toLowerCase()).whereIn("id", idsArr)
```

Comment:
Please try not to use any abbreviations like Arr if you can avoid it. Think of a better name. Why not just `populatingModels`? Since this is plural, it's enough to understand that we have an array.